zebedee-cms/src/main/java/com/github/onsdigital/zebedee/permissions/service/JWTPermissionsServiceImpl.java

```diff
@@ -0,0 +1,282 @@
+package com.github.onsdigital.zebedee.permissions.service;
+
+import com.github.onsdigital.exceptions.JWTVerificationException;
+import com.github.onsdigital.zebedee.json.CollectionDescription;
+import com.github.onsdigital.zebedee.json.PermissionDefinition;
+import com.github.onsdigital.zebedee.model.Collection;
+import com.github.onsdigital.zebedee.session.model.Session;
+import com.github.onsdigital.zebedee.session.service.Sessions;
+import com.github.onsdigital.zebedee.teams.model.Team;
+import com.github.onsdigital.zebedee.user.model.User;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import static com.github.onsdigital.zebedee.logging.CMSLogEvent.warn;
+
+public class JWTPermissionsServiceImpl implements PermissionsService {
+    static final String PUBLISHER_PERMISSIONS = "publisher";
+    static final String ADMIN_PERMISSIONS = "admin";
+    static final String JWTPERMISSIONSSERVICE_ERROR =
+            "error accessing JWTPermissions Service";
+    private Sessions sessionsService;
+    private Session session;
+
+
+    /**
+     * @param sessionService
+     */
+
+    public JWTPermissionsServiceImpl(Sessions sessionService) {
+        this.sessionsService = sessionsService;
+    }
+
+    /**
+     * @param session {@link Session} to get the user details from.
+     * @return boolean
+     * @throws JWTVerificationException
+     */
+    @Override
+    public boolean isPublisher(Session session) throws JWTVerificationException {
+        // Get JWT from JWT session service and check if the user has the 'Publisher' permission in their groups.
+        return session != null && !StringUtils.isEmpty(session.getEmail()) &&
+                hasPermission(session, PUBLISHER_PERMISSIONS);
+    }
+
+    /**
+     * @param email the email of the user to check.
+     * @return boolean
+     * @throws IOException
+     */
+    @Override
+    public boolean isPublisher(String email) throws IOException {
+        boolean result = true;
+        if (StringUtils.isEmpty(email)) {
```

Comment:
Logic is fine but I think we could refactor this to simplify it a bit and remove some of the nesting and else block:

```
if (StringUtils.isEmpty(email)) {
    warn().user(email).log("request for isPublisher unsuccessful empty email");
    return false;
} 

Session s = null;
try {
    Session s = getSessionfromEmail(email);
} catch (Exception ex) {
    warn().user(email).exception(ex).log("request for publisher privilages exception caught");
    returen false;
}

 return hasPermission(s, PUBLISHER_PERMISSIONS);
}
```